integration-cli/docker_cli_build_test.go

```diff
@@ -5438,3 +5438,282 @@ func (s *DockerSuite) TestBuildEmptyStringVolume(c *check.C) {
+		map[string]string{
+			addVal:  "some stuff",
+			copyVal: "some stuff",
+		})
+	defer ctx.Close()
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	args := []string{
+		"--build-env", fmt.Sprintf("%s=%s", wdVar, wdVal),
+		"--build-env", fmt.Sprintf("%s=%s", addVar, addVal),
+		"--build-env", fmt.Sprintf("%s=%s", copyVar, copyVal),
+		"--build-env", fmt.Sprintf("%s=%s", envVar, envVal),
+		"--build-env", fmt.Sprintf("%s=%s", exposeVar, exposeVal),
+		"--build-env", fmt.Sprintf("%s=%s", userVar, userVal),
+		"--build-env", fmt.Sprintf("%s=%s", volVar, volVal),
+	}
+	if _, err := buildImageFromContextWithArgs(imgName, ctx, args...); err != nil {
+		c.Fatal(err)
+	}
+
+	var resMap map[string]interface{}
+	var resArr []string
+	res := ""
+	res, err = inspectFieldJSON(imgName, "Config.WorkingDir")
+	if err != nil {
+		c.Fatal(err)
+	}
+	if strings.Trim(res, "\"") != wdVal {
+		c.Fatalf("Config.WorkingDir value mismatch. Expected: %s, got: %s", wdVal, res)
+	}
+
+	res, err = inspectFieldJSON(imgName, "Config.Env")
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	if err := json.Unmarshal([]byte(res), &resArr); err != nil {
+		c.Fatal(err)
+	}
+	found := false
+	for _, v := range resArr {
+		if fmt.Sprintf("%s=%s", envVar, envVal) == v {
+			found = true
+			break
+		}
+	}
+	if !found {
+		c.Fatalf("Config.Env value mismatch. Expected <key=value> to exist: %s=%s, got: %v",
+			envVar, envVal, resArr)
+	}
+
+	res, err = inspectFieldJSON(imgName, "Config.ExposedPorts")
+	if err != nil {
+		c.Fatal(err)
+	}
+	if err := json.Unmarshal([]byte(res), &resMap); err != nil {
+		c.Fatal(err)
+	}
+	if _, ok := resMap[fmt.Sprintf("%s/tcp", exposeVal)]; !ok {
+		c.Fatalf("Config.ExposedPorts value mismatch. Expected exposed port: %s/tcp, got: %v", exposeVal, resMap)
+	}
+
+	res, err = inspectFieldJSON(imgName, "Config.User")
+	if err != nil {
+		c.Fatal(err)
+	}
+	if strings.Trim(res, "\"") != userVal {
+		c.Fatalf("Config.User value mismatch. Expected: %s, got: %s", userVal, res)
+	}
+
+	res, err = inspectFieldJSON(imgName, "Config.Volumes")
+	if err != nil {
+		c.Fatal(err)
+	}
+	if err := json.Unmarshal([]byte(res), &resMap); err != nil {
+		c.Fatal(err)
+	}
+	if _, ok := resMap[volVal]; !ok {
+		c.Fatalf("Config.Volumes value mismatch. Expected volume: %s, got: %v", volVal, resMap)
+	}
+}
+
+func (s *DockerSuite) TestBuildBuildTimeEnvExpansionOverride(c *check.C) {
+	imgName := "bldvarstest"
+	defer func() { deleteImages(imgName) }()
+	envKey := "foo"
+	envVal := "bar"
+	envKey1 := "foo1"
+	envValOveride := "barOverride"
+	buildCmd := exec.Command(dockerBinary, "build", "-t", imgName, "--build-env",
+		fmt.Sprintf("%s=%s", envKey, envVal), "-")
+	buildCmd.Stdin = strings.NewReader(fmt.Sprintf("FROM busybox\n"+
+		"ENV %s %s \n"+
+		"ENV %s ${%s} \n"+
+		"RUN echo $%s\n"+
+		"CMD echo $%s\n",
+		envKey, envValOveride,
+		envKey1, envKey,
+		envKey1, envKey1))
+
+	if out, _, err := runCommandWithOutput(buildCmd); err != nil || !strings.Contains(out, envValOveride) {
```

Comment:
I think this will always return true because 'out' will always have envValOveride due to it appearing on the first ENV line, no?